### DISCUSSION TITLE
Strange behaviour when inserting a function inside paper attributes

### DISCUSSION THREAD

**Discussion by Biowavix** [clientIO/joint#2033] [Q&A] [answered]
Hello, 

Maybe this isn't a very good practise, but I find it very useful to have global attributes that all cells in jointJs have access to. I discovered that dia.paper can have attributes that all views have access to, so I was using it as a global variable container.

The problem comes when I try to store a function (I repeat that maybe is a very bad practise) inside the paper attributes. Apparently everything go well and I can do thins like:

`cellView.paper.attributes.myFunction(myValue)`

The problem is that when I define the paper,
```
  const paper = new joint.dia.Paper({
    attributes: {
      myFunction: (myValue) => {
        console.log("Do something with value: ",myValue);
      },
    },
    width: 100%,
    ...
```

the function that I have assigned is executed automatically, with a 0 as input parameter, that is to say, when creating the previous paper, without explicitly calling to myFunction, the console shows:

`>>> Do something with value: 0`

[Here](https://codepen.io/djimenez-developair/pen/gOjJxJe?editors=1111) is a codepen with a minimal example.

Any idea of why this happens?
Is there a cleaner alternative to do this?

Thank you in advance!

**Comment by kumilingus** [comment, marked as answer]
You are using `attributes` property of `Backbone.View`. See the [docs](https://backbonejs.org/#View-attributes).
If you rename it for instance to `attributes2` it will not get invoked.

**Reply by Biowavix** [reply]
understood and corrected, thank you!
